### ISSUE TITLE
DoReFa framework can't reproduce results of TF slim models

### ISSUE THREAD

**Issue by stecklin** [ppwwyyxx/tensorpack#426]
There was a similar issue here ([https://github.com/ppwwyyxx/tensorpack/issues/421](url)), but I am observing the problem with minimal changes to the original code:

In `resnet-dorefa.py` I simply replaced the model definition with `logits = mobilenet_v1(...)` from TF slim and ran an evaluation on imagenet as proposed in the main method. I didn't use any quantization (`BITW, BITA, BITG = 32`). Using the official checkpoint of MobileNet results in a top-1 error of 98%.

Hence, there seems to be an issue in the evaluation framework as provided in the resnet example.

**Comment by ppwwyyxx** [comment]
As I said in #421 already there are many places things can be different. Calling a model from slim doesn't make them equivalent.

Most likely your issue is because of this: https://github.com/ppwwyyxx/tensorpack/issues/148#issuecomment-285965752

**Comment by stecklin** [comment]
Thanks for the quick response. I was in fact missing to adjust the labels, results are much better now.
Yet still the performance is notably (-15%) worse than it should be. Are there more common pitfalls you know of? I paid attention to the value ranges and BGR/RGB space as previously mentioned by you.

**Comment by ppwwyyxx** [comment]
Crop, resize, mean subtraction, or any other possible preprocessing. Basically you need to go through a working evaluation code (assuming they have one) and figure out everything that's happening.
